### ISSUE TITLE
PLEASE STOP THIS PROJECT!

### ISSUE THREAD

**Issue by Tset-Noitamotua** [Microsoft/BashOnWindows#935]
Do us (Windows users) a favour and put this effort into improving your command prompt (cmd.exe) which is still lagging behind [cmder](http://cmder.net/) ([conemu](https://conemu.github.io/)) and other terminal emulators out there.

This pseudo Linux Bash just don't feel right - same as cygwin, babun and co. And even if you tried to make it feel right than you would have to reinvent Linux (on Windows!) - which I guess you don't intend to do. For Linux on Windows there are virtual machines (e.g. [Vagrant](https://www.vagrantup.com/)) and containers (e.g. [Docker](https://www.docker.com/)).

Just integrate some of the cool Linux commands seamlessly into Windows's terminal / command prompt and let Windows user have their native user experience. No need for a Linux subsystem or a Linux user on Windows, really  ;-)))

**Comment by SecurityWaffle** [comment]
Have you given PowerShell a try instead of always using CMD? A lot of the functionality (if not all) that you are referencing Windows is needing exists in PowerShell. I like seeing Bash on Windows as it allows me to natively run tools that I normally wouldn't be able to run on a Windows box. It's not designed to be a terminal emulator for Windows. They have a lot of work ahead of them but are making good progress.

**Comment by fpqc** [comment]
@Tset-Noitamotua This project actually has very little to do with "Bash on Windows" (imo it is a misnomer).  Bash on Windows is actually a sort of proof of concept for the larger Linux Subsystem.

Windows Console improvements are planned to take place in concert with the Linux subsystem project (WSL), but they are distinct projects inside of Microsoft (even though the teams do work together).  

If you noticed, @bitcrazed assigns console issues to @zadjii-msft and not to the other people on this project, and that's because he's the guy working on getting the Windows console working with the bash wrapper (although I don't know if he's formally part of the WSL team or the console team).

**Comment by ericblade** [comment]
I assumed that this was a bit of sarcasm, but since we're discussing it.. PowerShell's learning curve seems to be quite excessive for little obvious benefit to the casual user.

**Comment by iz0eyj** [comment]
I think it's time for a new model of development and implementation in which free software (and open source) rests on a solid, reliable and modern proprietary kernel code.
All the freedom to develop and deploy you want for your code, but full control of the operating system by two large and serious companies like Microsoft and Canonical, which eliminates at the root development anarchy and fragmentation.
With native support for Linux binaries, the two companies are doing just that: 'open' paradigm suitable to the future.
As it is currently implemented it guarantees countless possibilities of a merger on a single desktop of the best features of those which are now the two Windows 10 garments (the one open and one closed), but in the future I think it is conceivable a WSL autonomous, a MC/* nix (Microsoft-Canonical/* nix) standalone ... maybe based on something similar to the current Hyper-V Server , which is little more than a kernel stripped of everything that isn't necessary bearing layer above the WSL.
You could call it MCNux? :)
I hope that team members don't want to kill me for this post.

I'm sorry for my bad english.

**Comment by Tset-Noitamotua** [comment]
@SecurityWaffle  I always tried to avoid using powershell in the past but since I got in touch with Packer, Vagrant, Docker and the like I thing there is no way around to get familiar with it seriously  ... so it´s on my TODO list :)

And hey, I don´t mean what I wrote as evil as it might sound. As I first heard about this (lets call it by name) WSL (Windows Subsystem for Linux) my first thought was "Great! Finally I can get rid of all that different tools like Cmder, ConEmu, Cygwin, Babun, GitBash etc. and have everything in one place - in a Windows native "terminal" (I know it´s the wrong term) with the ability to execute Linux (like) commands". But after I got a little taste of it (as [described here](https://github.com/Microsoft/BashOnWindows/issues/932)) I am sure that I will stick with Cmder still a while because I like their approach more. I have almost everythink in one place even some linux-like commands and if I am missing something like e.g. `wget` - I just download the `wget.exe` for Windows, put it in Cmder's `bin` directory and there I go, it's on PATH and I can use it from everywhere on my native Windows filesystem. I don´t need to mess around with Linux-like folder structure (`/usr/local/bin/`) or to deal with two isolated PATHs (one on Windows and one on WSL) not to think about environment variables - with WSL one would have to manage all that stuff twice ... proxy settings, .git config, .ssh config  ... oh my god. I don´t even want to think about it - it´s a nightmare even without WSL :)

All that said, I think that any effort of bringing open source to Windows and vise versa and any enhencement of cross-platform development experience is worth it. So keep up the good work! (But don´t forget to pimp the command prompt ;-))))

**Comment by iz0eyj** [comment]
@Tset-Noitamotua I think you're misunderstanding the nature of WSL.
WSL is a Windows subsystem ... is part of it as are RT and Win32.
The novelty (and need) lies precisely in this: making Microsoft operating system even more interoperable with all that exists, pushing him into the border once considered impassable (almost there was an invisible Hadrian's wall) of Free Software.
He's still young, he's only a few months old, but its purpose is to make Windows 10 simultaneously owner and free, and make both sides equal.
Give it some time, enjoy the results, and if you can use your skills for aiding the debugging.

**Comment by barry-scott** [comment]
I would think that CMDis beyond recovery because of the need to run old scripts.
All of CMD's issues are too deeply in its core.
Take the problems with quoting strings and passing them around as one example.
If you fixed the quoting then you break existing scripts.

**Comment by Deiru2k** [comment]
Eh, this Issue is a definition of "jumping the gun".
The project has literraly nothing to do with the bash itself, rather, this is an attempt at establishing a solid, concrete bridge between linux and windows ecosystems, to allow windows users to benifit from linux in a ways including, but not limited to, bash shell.
This is the reverse-wine done right.

**Comment by FaraoneLele** [comment]
please NOT stop this project. it is beautifoul!!!

**Comment by fpqc** [comment]
@barry-scott There is no problem with `cmd.exe`, which is a shell rather than a virtual console/terminal emulator (that program is called the Windows Console Host (`conhost.exe`)).  Running cmd.exe in Windows actually opens up an interactive cmd.exe session running on conhost.

In theory, there is no reason why `cmd.exe` would have to be changed or removed, since it is just another shell for the Windows operating system.  The most feature-complete Windows shell is powershell, which also runs in `conhost.exe` and has deep connections with the Win32 API with native support for things like COM and registry management.  Because Windows uses file extensions to determine default applications to open things, batch scripts (`.BAT` files) could still open with cmd, while you could set bash.exe as the default program to use to open .sh or .bsh shellscripts from Windows (this doesn't exactly work since they need to be started with special arguments and with a copy of bash running in the background in interactive mode, but the concept works fine).

**Comment by iz0eyj** [comment]
@Deiru2k & @benhillis I think the project name, bash, is wrong because the people thinks to a simple command processor.

**Comment by fpqc** [comment]
@iz0eyj The official name (on the MSDN blog, in the Windows 10 component manager, etc.) is the Windows Subsystem for Linux, and in the internal Microsoft naming scheme (judging from certain filenames like lxss, lxsessionmanager, lxsession, etc), it apparently is called "lxss".

**Comment by Tset-Noitamotua** [comment]
Digging deeper I am beginning to understand that WSL is about much more
than just the bash shell. In that case the project's name indeed is a bit
missleading. So I am pretty sure I am not going to be the last one who was
trying to use this tool with wrong expectations.
Federico <notifications@github.com> schrieb am Fr., 19. Aug. 2016 um 20:00:

**Comment by Tset-Noitamotua** [comment]
Then Microsoft should avoid to promote it as BashOnWindows.

fpqc <notifications@github.com> schrieb am Fr., 19. Aug. 2016 um 20:22:

**Comment by fpqc** [comment]
@Tset-Noitamotua They called it "Bash On Windows" because that was the name of the feature that people requested when they first put up the Windows 10 User voice.  In fact, it's much much more than that, so you can't blame them for I guess ginning it up a little bit and giving it an exciting name, since in fact this is just a tiny simple application of the project, almost a proof of concept.

**Comment by iz0eyj** [comment]
@fpqc Yes, I know all... but for large oart of end-user this project is "bash", and this people thinks it is a simple shell.
Also WSL is wrong, this project needs a "true name", a logo and a changelog included in the main Win10 note of versiin.

**Comment by bitcrazed** [comment]
@Tset-Noitamotua "Digging deeper I am beginning to understand that WSL is about much more
than just the bash shell"

Glad you're finding that there's considerable value in what we're building. 

FWIW, we are working hard to overhaul the Windows Console itself. Console has received little love over the past few decades, but we've got a dedicated, passionate, full-time team working on a major overhaul as I type. Lots more detail to come, but know that there's a lot of good stuff coming to the console over the next few releases.

And re. the naming; we hear ya.

**Comment by SRGOM** [comment]
@ericblade Are you sure we have the right distinction here between shell, readline, coreutils, ssh etc. 

Most people club these together sbash. Bash is none of these. Bash is the super awful square brackets with spaces required for if while

The tools you see in powershell are the canonical way of doing things. You can continue to use coreutils and no one would judge you. Anecdotal but i had no issues starting in powershell and love it's predictability and specially the canonical verbose way of doing things in maintainable scripts.

**Comment by barry-scott** [comment]
@fpqc The details of the conhost vs shell do not have any bearing on the fitness of the shell hosted in it. You did not address the problem of quoted strings in CMD.EXE that makes it very hard to use robustly.

**Comment by blueben** [comment]
Why is this issue still open? It's clearly a wontfix.

**Comment by zadjii-msft** [comment]
@blueben Great point. I think it's safe to say this discussion has run it's course.

**Comment by rsenna** [comment]
TLDR; Guy comes out of nowhere, complaining that lxss is no better than Cmder. 1) lxss is much more than simply "Bash on Windows". 2) Even if it wasn't, it is clearly NOT another terminal emulator (like Cmder is).

**Comment by rsenna** [comment]
PLEASE DELETE THIS QUESTION (or at least rename it). It's absurd. OP clearly do not know the difference between a shell, a terminal emulator, and what Lxss stands for (also, it's the **first** hits when someone googles for "Lxss terminal emulator", instead of, say, https://github.com/goreliu/wsl-terminal).

**Comment by Tset-Noitamotua** [comment]
haha ... I wish It would be that easy to get "a first hit" on google with other things, too :)))

I have no pain to rename it. Tell me what title you want it to have. (Hope that it holds it's first position in google search after renaming)

**Comment by shoogle** [comment]
@Tset-Noitamotua, how about renaming it "Is WSL more than just Bash on Windows?"

**Comment by fpqc** [comment]
@shoogle they just did this, it's on the console blog

**Comment by heshamh96** [comment]
neither flatpak nor Snapd installations  are working ;like at
**** Failed to open file '/proc/sys/kernel/random/boot_id': No such file or directoryall ... and it gives fatal errors like (flatpak installation error)
or 
****error: cannot communicate with server: Post http://localhost/v2/snaps/telegram-latest: dial unix /run/snapd.socket: connect: no such file or directory  (Snapd installation error)
